pkg/devfile/parser/context.go

```diff
@@ -0,0 +1,69 @@
+package parser
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+)
+
+// DevfileCtx stores context info regarding devfile
+type DevfileCtx struct {
+	apiVersion string
+	absPath    string
+	relPath    string
+	rawContent []byte
+	jsonSchema string
+}
+
+// NewDevfileCtx returns a new DevfileCtx type object
+func NewDevfileCtx(path string) DevfileCtx {
+	return DevfileCtx{
+		relPath: path,
+	}
+}
+
+// Populate fills the DevfileCtx struct with relevant context info
+func (d *DevfileCtx) Populate() (err error) {
+
+	// Get devfile absolute path
+	if d.absPath, err = getAbsFilePath(d.relPath); err != nil {
+		return err
+	}
+	glog.V(4).Infof("Absolute devfile path: '%s'", d.absPath)
+
+	// Read and save devfile content
+	if err := d.SetDevfileContent(); err != nil {
+		return err
+	}
+
+	// Get devfile APIVersion
+	if err := d.SetDevfileAPIVersion(); err != nil {
+		return err
+	}
+
+	// Check if the apiVersion is supported
+	if !d.IsApiVersionSupported() {
+		errMsg := fmt.Sprintf("Devfile apiVersion '%s' is not supported by odo", d.apiVersion)
+		return fmt.Errorf(errMsg)
+	}
+	glog.V(4).Infof("Devfile apiVersion '%s' is supported in odo", d.apiVersion)
+
+	// Read and save devfile JSON schema for provided apiVersion
+	if err := d.SetDevfileJSONSchema(); err != nil {
+		return err
+	}
+
+	// Successfull
+	return nil
+}
+
+// Validate func validates devfile JSON schema for the given apiVersion
+func (d *DevfileCtx) Validate() error {
```

Comment:
is this function needed? seems like it only holds a function call with error check

Reply:
Done